components/mdns/examples/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+# The following lines of boilerplate have to be in your project's CMakeLists
+# in this exact order for cmake to work correctly
+cmake_minimum_required(VERSION 3.5)
+
+set(EXTRA_COMPONENT_DIRS "../..")
+# This example uses an extra component for common functions such as Wi-Fi and Ethernet connection.
+list(APPEND EXTRA_COMPONENT_DIRS $ENV{IDF_PATH}/examples/common_components/protocol_examples_common)
```

Comment:
I'd recommend not depending on protocol_examples_common in examples outside of IDF. The thing is, protocol_examples_common are not part of the public API. The functions defined by this component can be changed in backwards incompatible way in any version of IDF. This will make this component incompatible.